spot_driver/launch/spot_driver.launch.py

```diff
@@ -116,6 +116,11 @@ def launch_setup(context: LaunchContext, ld: LaunchDescription) -> None:
     publish_point_clouds_config = LaunchConfiguration("publish_point_clouds")
     mock_enable = IfCondition(LaunchConfiguration("mock_enable", default="False")).evaluate(context)
 
+    # if config_file has been set (and is not the default empty string) and is also not a file, do not launch anything.
+    config_file_path = config_file.perform(context)
+    if (config_file_path != "") and (not os.path.isfile(config_file_path)):
```

Comment:
I would like a better way of checking the default value of `config_file` launch argument to avoid hard-coding values, but unfortunately there is no method in the `LaunchConfiguration` class that does this.

Reply:
nit: do you know if checking `isfile` on an empty string error out or return false? if it returns false we could remove the first clause and just check if the file exists

Reply:
isfile on an empty string does return false. I opted  not to do that because there is the option to pass in parameters through environment variables instead of a config file. So if the user sets environment variables and chooses to not pass in a config file, they would not be able to run the launchfile anymore. Open to suggestions on better ways to handle this though